lib/src/simulations/maurer_rose.dart

```diff
@@ -0,0 +1,388 @@
+    @required this.thickness,
+  })  : _d = d,
+        _n = n,
+        super(key: key);
+
+  final double _d;
+  final double _n;
+  final bool animate;
+  final bool animating;
+  final double thickness;
+
+  @override
+  MaurerRoseState createState() => MaurerRoseState();
+}
+
+class MaurerRoseState extends State<MaurerRose> {
+  List<Offset> points = [];
+  List<Offset> points2 = [];
+  double loopi = 0;
+  double loopi2 = 0;
+  double r, n, d, c, transformx, transformy;
+  double looplength = 360;
+  double looplength2 = 360;
+
+  void dispose() {
+    super.dispose();
+  }
+
+  void clearscreen() {
+    points.clear();
+    points2.clear();
+    looplength = 360;
+    looplength2 = 360;
+    looplength += loopi;
+    looplength2 += loopi2;
+  }
+
+  nextStep() {
+    if (loopi >= looplength) {
+      clearscreen();
+      loopi = 0;
+      loopi2 = 0;
+      looplength = 360;
+      looplength2 = 360;
+    }
+    setState(() {
+      sleep(Duration(milliseconds: 10));
+      loopi += 1;
+      
+      points.add(Offset(300*sin(widget._n*loopi*widget._d*pi/180)*cos(loopi*widget._d*pi/180)/2,300*sin(widget._n*loopi*widget._d*pi/180)*sin(loopi*widget._d*pi/180)/2)
+          .translate((MediaQuery.of(context).size.width / 2).roundToDouble(),
+              (MediaQuery.of(context).size.height / 3).roundToDouble()));
+
+              sleep(Duration(milliseconds: 10));
+      loopi2 += 1;
+      points2.add(Offset(300*sin(widget._n*loopi2*pi/180)*cos(loopi2*pi/180)/2,300*sin(widget._n*loopi2*pi/180)*sin(loopi2*pi/180)/2)
+          .translate((MediaQuery.of(context).size.width / 2).roundToDouble(),
+              (MediaQuery.of(context).size.height / 3).roundToDouble()));
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (widget.animating) {
+      WidgetsBinding.instance.addPostFrameCallback((_) {
+        nextStep();
+      });
+    }
+    return CustomPaint(
+      painter: MaurerRosePainter(
+        widget._d,
+        widget._n,
+        (MediaQuery.of(context).size.width / 2).roundToDouble(),
+        (MediaQuery.of(context).size.height / 3).roundToDouble(),
+        
+        
+        widget.animate,
+        points,
+        points2,
+        widget.thickness,
+      ),
+      child: Container(),
+    );
+  }
+}
+
+class MaurerRosePainter extends CustomPainter {
+  double d, n, c;
+  double k, transformx, transformy;
+  List<Offset> points = [];
+  List<Offset> points2 = [];
+  bool animate;
+  double thickness;
+  MaurerRosePainter(
+    this.d,
+    this.n,
+    this.transformx,
+    this.transformy,
+    
+    
+    this.animate,
+    points,
+    points2,
+    this.thickness,
+  ) {
+    this.points = new List<Offset>.from(points);
+    this.points2 = new List<Offset>.from(points2);
+    k = n / d;
+  }
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    var paint = Paint()
+    ..color = Colors.black
+    ..style = PaintingStyle.stroke
+    ..strokeWidth = thickness;
```

Comment:
There's no need to keep the thickness of the inner lines the same as the outer curve. let's just keep it constant to 1?